common/src/main/java/org/opensearch/sdk/SdkClientImpl.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.sdk;
+
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executor;
+
+public abstract class SdkClientImpl {
```

Comment:
This should be an interface instead of abstract class

Reply:
Rename to SDKClientInterface?

Reply:
SdkClientInterface comes from the Department of Redundancy Department. :)

I did SdkClientDelegate.